app/services/worker_monitor.py

```
@@ -0,0 +1,703 @@
+            await self._save_worker(worker)
+
+    async def _cleanup_loop(self) -> None:
+        """Periodic cleanup of stale workers"""
+        while self._running:
+            try:
+                await asyncio.sleep(self.config.cleanup_interval)
+                await self._cleanup_offline_workers()
+                await self._update_worker_statuses()
+            except asyncio.CancelledError:
+                break
+            except Exception as e:
```

Ruff (0.14.14)

Do not catch blind exception: `Exception`

(BLE001)